src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -1984,11 +2019,18 @@ private List<string> EnvironmentOverrides(List<string> local, params string[] en
 
         private string EnvironmentOverrides(string local, params string[] environmentVariableNames)
         {
-            return (environmentVariableNames ?? Enumerable.Empty<string>())
+            var envValue = (environmentVariableNames ?? Enumerable.Empty<string>())
                 .Select(_environment.GetEnvironmentVariable)
                 .Where(value => value != null)
-                .FirstOrDefault()
-                ?? local;
+                .FirstOrDefault();
+
+            // prevent whitespace from leaking through
+            if (string.IsNullOrWhiteSpace(envValue))
```

Comment:
Thoughts/feelings about trimming the value we read? I can't think of any use cases where we'd want extra whitespace at the start/end of env vars

Reply:
Yeah, I was waffling back and forth on using trim vs just detecting.  I am fine using trim, what do you think?

Reply:
I think trimming makes sense.

Reply:
I agree, that we should trim this. I can only see it helping customers